packages/synapse-react-client/src/components/widgets/facet-nav/PlotsContainer.tsx

```diff
@@ -0,0 +1,379 @@
+  }
+}
+
+function convertPlotlyPlotTypeToFacetNavPlotType(
+  plotlyPlotType: PlotlyPlotType,
+): PlotType {
+  if (plotlyPlotType === 'bar') {
+    return 'BAR'
+  } else if (plotlyPlotType === 'pie') {
+    return 'PIE'
+  }
+  return 'PIE'
+}
+
+const generatePlotKey = (plotUiState: UiPlotState) => {
+  if ('__custom' in plotUiState.plotId) {
+    // For custom plots
+    return `custom-${plotUiState.plotId.title}`
+  } else {
+    // For facet plots
+    return `facet-${plotUiState.plotId.columnName}-${plotUiState.plotId.jsonPath}`
+  }
+}
+
+const DEFAULT_PLOT_TYPE: PlotType = 'PIE'
+
+const DEFAULT_FACETS_TO_PLOT: string[] = []
+const DEFAULT_CUSTOM_PLOTS: QueryWrapperSynapsePlotProps[] = []
+
+export default function PlotsContainer(props: PlotsContainerProps) {
+  const {
+    facetsToPlot = DEFAULT_FACETS_TO_PLOT,
+    customPlots = DEFAULT_CUSTOM_PLOTS,
+  } = props
+  const { error } = useQueryContext()
+  const data = useAtomValue(tableQueryDataAtom)
+  const isLoadingNewBundle = useAtomValue(isLoadingNewBundleAtom)
+  const { showPlots: showPlotVisualization } = useQueryVisualizationContext()
+  const [plotUiStateArray, setPlotUiStateArray] = useState<UiPlotState[]>([])
+  const facetNavPanelPropsArray = useFacetPlots(facetsToPlot)
+
+  const getPlotType = (plotId: PlotIdentifier): PlotType => {
+    const plotType = plotUiStateArray.find(item =>
+      plotMatchesDefinition(plotId, item.plotId),
+    )?.plotType
+    return plotType ?? 'PIE'
+  }
+
+  // Wrap in useCallback
+  // switch to
+  const isPlotInState = useCallback(
+    (plotId: PlotIdentifier): boolean => {
+      // plotUiStateArray.find...
+      //return plotUiStateArray && plotUiStateArray.find(plotId)
+      return !!plotUiStateArray.find(plot => {
+        if ('__custom' in plot.plotId && '__custom' in plotId) {
+          return plot.plotId.title === plotId.title
+        } else if (!('__custom' in plotId) && !('__custom' in plot.plotId)) {
+          return facetObjectMatchesDefinition(plot.plotId, plotId)
+        } else {
+          return false
+        }
+      })
+    },
+    [plotUiStateArray],
+  )
+
+  // Wrap in useCallback
+  const getCombinedNewPlots = useCallback(
+    (
+      customPlots: QueryWrapperSynapsePlotProps[],
+      facetNavPanelPropsArray: Pick<
+        FacetNavPanelProps,
+        'applyChangesToFacetFilter' | 'applyChangesToGraphSlice' | 'facetToPlot'
+      >[],
+    ): UiPlotState[] => [
+      ...(customPlots ?? []).map((plotProps, index) => ({
+        plotId: {
+          __custom: true,
+          title: plotProps.title ?? '',
+        } satisfies CustomPlotIdentifier,
+        isHidden: index >= DEFAULT_VISIBLE_PLOTS,
+        plotType: convertPlotlyPlotTypeToFacetNavPlotType(plotProps.type),
+      })),
+      ...(facetNavPanelPropsArray ?? []).map((facetPlotProps, index) => ({
+        plotId: facetPlotProps.facetToPlot,
+        isHidden: index >= DEFAULT_VISIBLE_PLOTS,
+        plotType: DEFAULT_PLOT_TYPE,
+      })),
```

Comment:
Need to use same index to show 2 plots total, not 2 of each
```suggestion
      ...(customPlots ?? []).map((plotProps, index) => ({
        plotId: {
          __custom: true,
          title: plotProps.title ?? '',
        } satisfies CustomPlotIdentifier,
        isHidden: index >= DEFAULT_VISIBLE_PLOTS,
        plotType: convertPlotlyPlotTypeToFacetNavPlotType(plotProps.type),
      })),
      ...(facetNavPanelPropsArray ?? []).map((facetPlotProps, index) => ({
        plotId: facetPlotProps.facetToPlot,
        isHidden: (index + (customPlots ?? []) .length) >= DEFAULT_VISIBLE_PLOTS,
        plotType: DEFAULT_PLOT_TYPE,
      })),
```